### PR TITLE
Add changelog for release 0.4.0

### PR DESCRIPTION
**Pull request by MathieuMorrissette**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+All notable changes to this project will be documented in this file.
+
+## [Unreleased]
+
+## [0.4.0] - 2020-03-15
+
+### Added
+#### General
+ - Symmetric encryption. (Rust, C#, JS)
```
Comment (pdugre) [inline CHANGELOG.md]: Asymmetric

File: CHANGELOG.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+   - Confusion between bits and bytes. The keys generated by the function would be 256 byte instead of 32 byte. The error is now fixed and our decryption will fallback to 256 bytes if the decryption fail.
+ - ```DevolutionsCryptoExceptions``` have now more details in the message and the stack trace.
+
+### Breaking Changes
+#### C#
+ - Renamed the following function
+   - ```GenerateKeyExchange``` to ```GenerateKeyPair```
+ - Renamed the following class
+   - ```KeyExchange``` to ```KeyPair```
+ - The function ```DecryptWithKey``` had a ```iterations``` parameter that wasn't used so it has been removed.
```
Comment (pdugre) [inline CHANGELOG.md]: Don't forget to talk about the Length default that was changed from 256 to 32 in DeriveKey

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+The CI now ensures that code pushed to the project is respecting our standards. It also prevent merge conflicts caused by the code formatting.
+
+##### C#
+ - Uses SecurityCodeScan and StylecopAnalyzers for code formatting checks.
+ - Warnings are now treated as errors.
+
+##### Rust
+ - Uses rustfmt and rust-clippy for code formatting checks. 
+
```
Comment (pdugre) [inline CHANGELOG.md]: JS part

File: CHANGELOG.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+   - ```GenerateKeyExchange``` to ```GenerateKeyPair```
+ - Renamed the following class
+   - ```KeyExchange``` to ```KeyPair```
+ - The function ```DecryptWithKey``` had a ```iterations``` parameter that wasn't used so it has been removed.
+
+### Unit Testing
+#### General
+ - Added unit tests for
+   - Symmetric encryption (C#, Rust, JS)
+   - Derive key pair from a password. (C#, Rust, JS)
```
Comment (pdugre) [inline CHANGELOG.md]: Asymmetric
Hashing
Secret Sharing(no c#)

Conformity Tests

File: CHANGELOG.md [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+##### C#
+ - Uses SecurityCodeScan and StylecopAnalyzers for code formatting checks.
+ - Warnings are now treated as errors.
+
+##### Rust
+ - Uses rustfmt and rust-clippy for code formatting checks. 
+
+##### Javascript
+ - Tests uses eslint for formatting, using [JavaScript Standard Style](https://standardjs.com/)
```
Comment (ekse) [inline CHANGELOG.md]: nit: Tests use

**Pull request merged**